state/processor.go

```diff
@@ -114,20 +168,42 @@ func (tp *TransactionProcessor) coalesceTransactionsBySender(transactions Transa
 	return trnsBySender
 }
 
-func (tp *TransactionProcessor) Process(transactions Transactions, trnsBySender map[common.Address][]*Transaction) error{
-	bySender := make(map[common.Hash]bool)
+func (tp *TransactionProcessor) Process(transactions Transactions, trnsBySender map[common.Address][]*Transaction) {
+	senderPut := make(map[common.Address]bool)
+	sortedOriginByTransactions := make([]common.Address, 0,10)
+
+	// The order of the transactions determines the order addresses by which we take transactions
+	// Maybe refactor this
 	for _, trans := range transactions {
-		for _, trns := range trnsBySender[trans.Origin] {
-			if _, ok := bySender[trns.Hash()]; !ok {
-				bySender[trans.Hash()] = true
-				err := tp.ApplyTransaction(trans)
-				if err != nil {
-					return err
-				}
+		if _, ok := senderPut[trans.Origin]; !ok {
+			sortedOriginByTransactions = append(sortedOriginByTransactions, trans.Origin)
+			senderPut[trans.Origin] =true
+		}
+	}
+
+
+	for _, origin := range sortedOriginByTransactions {
+		for _, trns := range trnsBySender[origin] {
+			//todo: should we abort all transaction processing if we failed this one?
+			err := tp.ApplyTransaction(trns)
+			if err != nil {
+				log.Error("transaction aborted: %v", err)
```

Comment:
maybe count successful processes and return the number ? so 0 might help to not move forward